src/Build/Evaluation/Evaluator.cs

```diff
@@ -523,47 +514,24 @@ private static ProjectTargetInstance ReadNewTargetElement(ProjectTargetElement t
             {
                 using (evaluationProfiler.TrackElement(targetChildElement))
                 {
-                    ProjectTaskElement task = targetChildElement as ProjectTaskElement;
-
-                    if (task != null)
-                    {
-                        ProjectTaskInstance taskInstance = ReadTaskElement(task);
-
-                        targetChildren.Add(taskInstance);
-                        continue;
-                    }
-
-                    ProjectPropertyGroupElement propertyGroup = targetChildElement as ProjectPropertyGroupElement;
-
-                    if (propertyGroup != null)
-                    {
-                        ProjectPropertyGroupTaskInstance propertyGroupInstance = ReadPropertyGroupUnderTargetElement(propertyGroup);
-
-                        targetChildren.Add(propertyGroupInstance);
-                        continue;
-                    }
-
-                    ProjectItemGroupElement itemGroup = targetChildElement as ProjectItemGroupElement;
-
-                    if (itemGroup != null)
-                    {
-                        ProjectItemGroupTaskInstance itemGroupInstance = ReadItemGroupUnderTargetElement(itemGroup);
-
-                        targetChildren.Add(itemGroupInstance);
-                        continue;
-                    }
-
-                    ProjectOnErrorElement onError = targetChildElement as ProjectOnErrorElement;
-
-                    if (onError != null)
+                    switch (targetChildElement)
```

Comment:
Can this be a switch expression?

```c#
targetChildren.Add(targetChildElement switch
{
    ProjectTaskElement task => ReadTaskElement(task),
    ProjectPropertyGroupElement propertyGroup => ReadPropertyGroupUnderTargetElement(propertyGroup),
    ProjectItemGroupElement itemGroup => ReadItemGroupUnderTargetElement(itemGroup),
    ProjectOnErrorElement onError => ReadOnErrorElement(onError),
    _ => throw ErrorUtilities.ThrowInternalError("Unexpected child")
});
```

Reply:
Looks like maybe not since the OnError case adds to a different list?